pkg/investigations/cannotretrieveupdatessre/cannotretrieveupdatessre.go

```diff
@@ -0,0 +1,164 @@
+		notes.AppendWarning("Alert escalated to on-call primary for review.")
+		logging.Infof("Escalating incident with notes for cluster %s", r.Cluster.ID())
+		err = r.PdClient.EscalateIncidentWithNote(notes.String())
+		if err != nil {
+			logging.Errorf("Failed to escalate incident to PagerDuty: %v", err)
+			return result, fmt.Errorf("failed to escalate incident: %w", err)
+		}
+		return result, err
+	}
+	if note != "" {
+		notes.AppendWarning(note)
+		err = r.PdClient.AddNote(notes.String())
+		if err != nil {
+			logging.Error("could not add notes to the incident")
+		}
+	}
+	if clusterVersion != "" {
+		notes.AppendSuccess("ClusterVersion found: %s", clusterVersion)
+		err = r.PdClient.AddNote(notes.String())
+		if err != nil {
+			logging.Error("could not add passed message to incident notes")
+		}
+	}
+
+	notes.AppendWarning("Alert escalated to on-call primary for review.")
+	logging.Infof("Escalating incident with notes for cluster %s", r.Cluster.ID())
+	err = r.PdClient.EscalateIncidentWithNote(notes.String())
+	if err != nil {
+		logging.Errorf("Failed to escalate incident to PagerDuty: %v", err)
+		return result, fmt.Errorf("failed to escalate incident: %w", err)
+	}
+	logging.Infof("Investigation completed and escalated successfully for cluster %s", r.Cluster.ID())
+
+	return result, nil
+}
+
+// checkClusterVersion retrieves the cluster version
+func checkClusterVersion(k8scli client.Client, clusterID string) (version string, note string, err error) {
+	logging.Infof("Checking ClusterVersion for cluster %s", clusterID)
+	clusterVersion := &configv1.ClusterVersion{}
+	err = k8scli.Get(context.TODO(), client.ObjectKey{Name: "version"}, clusterVersion)
+	if err != nil {
+		return "", "Failed to get ClusterVersion: cluster access issues detected", fmt.Errorf("failed to get ClusterVersion: %w", err)
+	}
+	logging.Infof("ClusterVersion channel: %s", clusterVersion.Spec.Channel)
+	logging.Infof("ClusterVersion found: %s", clusterVersion.Status.Desired.Version)
+	logging.Debugf("ClusterVersion conditions: %+v", clusterVersion.Status.Conditions)
+
+	for _, condition := range clusterVersion.Status.Conditions {
+		logging.Debugf("Checking ClusterVersion condition: Type=%s, Status=%s, Reason=%s, Message=%q",
+			condition.Type, condition.Status, condition.Reason, condition.Message)
+		if condition.Type == "RetrievedUpdates" && condition.Status == "False" {
+			if (condition.Reason == "VersionNotFound" || condition.Reason == "RemoteFailed") &&
+				strings.Contains(strings.TrimSpace(condition.Message), "Unable to retrieve available updates") {
+				logging.Warnf("Detected ClusterVersion error: Reason=%s, Message=%s", condition.Reason, condition.Message)
+				return "", fmt.Sprintf("ClusterVersion error detected: %s. Current version %s not found in channel %s",
+						condition.Message, clusterVersion.Status.Desired.Version, clusterVersion.Spec.Channel),
+					fmt.Errorf("clusterversion validation failed: %s", condition.Reason)
```

Comment:
I find it a tad bit confusing that we're returning an error here: it's not the state we want to see the `clusterversion` in, but I'd expect an error return to mean that we _couldn't_ check the `clusterversion`, not that we were able to check the clusterversion but didn't like the state we saw it in